syntax= "proto2";

service Rover {

  rpc GetMap (MapRequest) returns (MapResponse) {}

  rpc GetStreamOfCommands (MovesRequest) returns (MovesResponse) {}

  rpc GetMineSerialNumber (MineSerialNumRequest) returns (MineSerialNumResponse) {}

  rpc SayCompletedCommands (CompletedCommandsRequest) returns (CompletedCommandsResponse) {}

  rpc ShareMinePin (ShareMinePinRequest) returns (ShareMinePinResponse) {}
}

// 1
message MapRequest {
  required string rover_name = 1;
}

message MapResponse {
  required string map = 1;
}

// 2
message MovesRequest {
  required string rover_index = 1;
}

message MovesResponse {
  required string moves = 1;
}

// 3
message MineSerialNumRequest {
  required string rover_index = 1; // optional?
  required string mine_coord = 2;
}

message MineSerialNumResponse {
  required string rover_index = 1; // optional??
  required string mine_serial_num = 2;
}

// 4
message CompletedCommandsRequest {
  required string rover_index = 1;
}

message CompletedCommandsResponse {
  required string message = 1;
}

// 5
message ShareMinePinRequest {
  required string rover_index = 1; // optional??
  required string mine_pin = 2;
}

message ShareMinePinResponse {
  required string message = 1;
}




